src/Storage/Device/S3.php

```diff
@@ -249,12 +371,70 @@ public function move(string $source, string $target): bool
     public function delete(string $path, bool $recursive = false): bool
     {
         $uri = ($path !== '') ? '/' . \str_replace('%2F', '/', \rawurlencode($path)) : '/';
-
+        
+        unset($this->headers['content-type']);
+        $this->headers['content-md5'] = \base64_encode(md5('', true));
         $this->call(self::METHOD_DELETE, $uri);
 
         return true;
     }
 
+    private function listObjects($prefix = '', $maxKeys = 1000, $continuationToken = '')
+    {
+        $uri = '/';
+        $this->headers['content-type'] = 'text/plain';
+        $this->headers['content-md5'] = \base64_encode(md5('', true));
+
+        $parameters = [
+            'list-type' => 2,
+            'prefix' => $prefix,
+            'max-keys' => $maxKeys,
+        ];
+        if(!empty($continuationToken)) {
+            $parameters['continuation-token'] = $continuationToken;
+        }
+        $response = $this->call(self::METHOD_GET, $uri, '', $parameters);
+        return $response->body;
+    }
+
+    /**
+     * Delete files in given path, path must be a directory. Return true on success and false on failure.
+     *
+     * @see http://php.net/manual/en/function.filesize.php
+     *
+     * @param string $path
+     *
+     * @return bool
+     */
+    public function deletePath(string $path): bool
+    {
+        $uri = '/';
+        $continuationToken = '';
+        do {
+            $objects = $this->listObjects($path, continuationToken: $continuationToken);
+            $count = (int) ($objects['KeyCount'] ?? 1);
+            if($count < 1) {
+                break;
+            }
+            $continuationToken = $objects['NextContinuationToken'] ?? '';
+            $body = '<Delete xmlns="http://s3.amazonaws.com/doc/2006-03-01/">';
```

Comment:
Would it make more sense to use [simpleXML](https://www.php.net/simplexml) to build it rather than manipulating the strings directly ? This is very prone to human errors. WDYT ?

Reply:
I know only how to use simpleXML to parse XLM strings to object, don't know how I can build an XML as we need here using simpleXML, if you know how let's have a call and do it together.